### DISCUSSION TITLE
10 GBE lan network  and latency?

### DISCUSSION THREAD

**Discussion by kastorjunior** [apohl79/audiogridder#392] [Q&A]
Hello

I would like to know if having two 10 gbe network cards ( one on the production pc and one on the server) will improve plugin latency or if it is only the CPU power that makes the difference.
Is it possible that the server calculates the latency of the plugins to reduce it as much as possible on the production PC? For example to use the plugins in recording mode. I use thunderbolt soundcards.
In any case, I'm planning to create a 10 gbe network with this type of card. Thanks for your help. Good luck.
https://fr.shopping.rakuten.com/offer/buy/2159847807/asus-xg-c100c-adaptateur-reseau.html?bbaid=3250117092&t=3590029

**Comment by apohl79** [comment]
There are different factors, that can affect the network latency. The nic is definitely one but also the distance between computers and the number of network components in between (switches, routers).

The server is already measuring the processing time of plugins as well as the AG plugin. You can find statistics in the log files and you can watch realtime aggregations in the statistics windows. You can also use the plugin monitor to get insights into the performance of the remote plugins including the network latency.

That said, the server can't optimize the plugin performance. Also not sure what you mean by using plugins in recording mode.

**Comment by shittymodernart** [comment]
@kastorjunior Just curious how this ended up for you - did updating your nics improve your latency? I currently have a 10gbe nic on my mac mini and 1gbe on my windows server. I'm seeing a max of around 200 mb/s traffic and only around 30% CPU usage on the server, but I'm getting around 30ms of network latency and have to set my buffers to 3 or 4 blocks to prevent pops and clicks (processing latency is around 10ms, even in the most demanding projects). In either case, I just ordered a 2.5gbe adapter for the windows server because it was only $20 and I figured I'd see if it would help improve the network latency at all.  As a side note, I was connecting the two computers directly with a cat 5 cable and getting a max bandwidth of around 65-80mb/s, but then I connected them with a cat 8 cable and was able to achieve the 200 mb/s speeds.

**Comment by jaacMusic** [comment]
Quick note, having a faster throughput/connection doesn't improve latency but is great when trying to accomplish more audio channels! What affects latency are things like, CPU, network hops, frame size, network interface quality, and other high QoS priority items that are sent on the same connections (ex. VoIP).
I usually connect directly from my DAW to my AG server box on a separate NIC card (I tend to prefer server-grade NIC cards on both systems).
Just my two cents!

**Reply by shittymodernart** [reply]
Hey @jaacMusic, thanks for the two cents! I'm not sure if I should start a new post on this (I'm guessing there's probably already quite a few but I haven't been able to find them using "search"), but I'll post here for now.

I don't understand why Logic or Audiogridder (or a combination of the two) are struggling in every large project I have. I'll try to give as much info as possible, just disregard if anything's irrelevant.

- The host computer is a Mac M1 running Logic Pro 10.7.4 in Rosetta mode.
- I have a CAT 8 crossover cable going out from the 1gbe port directly into a dedicated 1/2.5/10g NIC on my Windows 10 server machine. I know the CAT 8 is overkill, seeing as the max throughput I could hope for is 1gb/s due to the limitation of my M1 nic. The only reason I got a CAT 8 was in the course of trying to blindly troubleshoot this issue.

The issue: 

- Whenever I'm running a large project with many AG plugins (There are about 20 in this specific project) I will constantly get these system overload notifications:

<img width="258" alt="System Overload Notification" src="https://user-images.githubusercontent.com/114954191/195404665-ffb1d07a-2437-4ee2-ac11-4738f6458767.png">

- The only way to get rid of the constant notifications is to increase the AG buffer size. I use the largest buffer size in Logic to mix (1024 samples). At this point in my project (with 20ish AG plugins), the only way I can get it to even play is to set the AG buffer to 8 blocks (16,384 samples), and even then Logic gives me the system overload notification after playing for 15-20 seconds. If I try to increase the buffer to 9 blocks, everything falls apart and the audio running through AG is no longer in sync with audio not processed through AG. 
- If I was maxing out my bandwidth and maxing my server processor, I'd understand that I've simply reached the limits of the technology. However, while processing audio i can observe that the bandwidth from AG (on the dedicated NIC) is only around 100-150Mbps, and the server CPU load is less than 50%.

![CPU while playing 186 buffer](https://user-images.githubusercontent.com/114954191/195406275-5d1ccb7c-661f-4d35-8495-b96d632c8b40.JPG)


<img width="637" alt="AGridder Monitor" src="https://user-images.githubusercontent.com/114954191/195406621-88fe4dfd-445e-4edf-8860-8b3ae2830da3.png">


- When I do an offline bounce, however, my bandwidth goes up to around 250Mbps and my server CPU usage is around 70%.

![CPU while bouncing 186 buffer](https://user-images.githubusercontent.com/114954191/195406561-1420d549-87d8-47df-9ded-4da2b9816ade.JPG)

So the obvious question here is this: What is limiting AG from utilizing higher bandwidth and more processor power to keep up with the project? Why do I even need to have a buffer at all if I am only using 1/10th of my available bandwidth and less than half of my server's processing power?

I'm posting this in the hope that this is just some stupid setting I have to change, either on my NICs or in AG to fully unlock the capability of the server machine. Or maybe I'm misunderstanding, and throwing more CPU at AG wouldn't really change anything (but why?)?

**Reply by apohl79** [reply]
The problem here is not network throughput but processing latency. There are a few chains (the ones in yellow), that seem to be running at the upper end of the time window available to process a sample block. They are probably going red some times too. If this happens too often, AGs buffer queue will run empty and block logic, which will cause the error you saw. Larger buffers will only delay the issue, if the processing chain is constantly too slow.

If you enable the realtime mode in the AG plugin (under Transfer Audio/MIDI), logic should not be showing the error, but you will see read errors in the plugin monitor. You can try to break your heavy chains into multiple AG plugins (as you have enough bandwidth). I heard that this can help on Pro Tools at least.

**Reply by apohl79** [reply]
You can try larger block sizes now with b15, which might help those higher latency chains. Fixed outbound buffers should be enabled.

**Reply by shittymodernart** [reply]
@apohl79 Thanks! I started a new project yesterday and kept all the AG plugin containers to having one plugin each and it seems to be working much, much smoother. I also noticed at one point I was passing over 300Mbps, which is higher than it ever was previously. Both of my network cards support jumbo packets, up to 9k. Is that something you recommend turning on? I read mixed things about it online; it sounds like on one hand it allows you to pass more data quicker, but on the other hand it slightly increases latency.
